pkg/core/blockchain.go

```diff
@@ -988,31 +954,35 @@ func (bc *Blockchain) HasBlock(hash util.Uint256) bool {
 }
 
 // CurrentBlockHash returns the highest processed block hash.
-func (bc *Blockchain) CurrentBlockHash() (hash util.Uint256) {
-	bc.headersOp <- func(headerList *HeaderHashList) {
-		hash = headerList.Get(int(bc.BlockHeight()))
+func (bc *Blockchain) CurrentBlockHash() util.Uint256 {
+	topBlock := bc.topBlock.Load()
+	if topBlock != nil {
+		if tb, ok := topBlock.(*block.Block); ok {
```

Comment:
Can it ever be `ok`?

Reply:
It can't, we only store valid Block pointers there.